source/ota.c

```diff
@@ -3113,26 +3110,36 @@ OtaErr_t OTA_Init( OtaAppBuffer_t * pOtaBuffer,
          */
         ( void ) otaAgent.pOtaInterface->os.event.init( NULL );
 
-        if( pThingName == NULL )
+        if( pOtaBuffer == NULL )
         {
-            LogError( ( "Error: Thing name is NULL.\r\n" ) );
+            LogError( ( "Error: pOtaBuffer is NULL.\r\n" ) );
         }
         else
         {
-            uint32_t strLength = ( uint32_t ) ( strlen( ( const char * ) pThingName ) );
+            /* Initialize application buffers. */
+            initializeAppBuffers( pOtaBuffer );
 
-            if( strLength <= otaconfigMAX_THINGNAME_LEN )
+            if( pThingName == NULL )
             {
-                /*
-                 * Store the Thing name to be used for topics later. Include zero terminator
-                 * when saving the Thing name.
-                 */
-                ( void ) memcpy( otaAgent.pThingName, pThingName, strLength + 1UL );
-                returnStatus = OtaErrNone;
+                LogError( ( "Error: Thing name is NULL.\r\n" ) );
```

Comment:
```suggestion
                LogError( ( "Error: Thing name is NULL." ) );
```
LogError messages don't include an \r\n in the rest of this file.